### DISCUSSION TITLE
Workaround for imposing different boundary conditions on the left vs right boundaries.

### DISCUSSION THREAD

**Discussion by ckming102** [jipolanco/BSplineKit.jl#69] [Q&A] [answered]
First off, thanks so much for this package. Really easy to get started with.

Right now, I am trying to create a `hybrid` basis using different recombination matrices from different boundary condition types; eg. Dirichlet on the left and Neumann on the right. In principle, it looks do-able by constructing a new recombination matrix using the different parts of the recombination matrices. But it does not seem like there is a straightforward way to do this right now with just the currently implemented module functions.

**Comment by jipolanco** [comment]
Thanks for your interest in this package!

Indeed currently there's no easy way of imposing different boundary conditions on each side. This is something that I've been wanting to do (shouldn't be very difficult). I'll try to take a look at this soon.

**Comment by jipolanco** [comment, marked as answer]
It should be now possible on v0.16 :) See also #70.

You can now create a recombined basis like this:

```julia
B = BSplineBasis(...)
R = RecombinedBSplineBasis(B, Derivative(0), Derivative(1))
```

where `Derivative(0)` means imposing Dirichlet conditions on the left, and `Derivative(1)` corresponds to Neumann on the right.

Let me know if you find any issues!

**Reply by ckming102** [reply]
Woah, you are really quick :) 

It works great actually. I just only updated from Pkg and tried it out. 

But will the re-ordering of ` RecombinedBSplineBasis()` method signature be a breaking change for other packages  using `BSplineKit.jl`? I am using this with `DifferentialEquations.jl` like in the heat-eqn example.

Thanks again!

**Reply by ckming102** [reply]
Oh it looks like, `BSplineKit.jl` isn't a dependency of `DifferentialEquations.jl`. But there are unsatisfiable constraints with versions of `FillArrays` and `BandedMatrices`, when adding `DifferentialEquations.jl` after the latest `BSplineKit.jl`. 

Edit: A simple work-around would to use `OrdinaryDiffEq.jl` directly which implements the `SecondODEProblem()`. You might want to update the heat-equation example in the `gh` pages.

**Reply by jipolanco** [reply]
I'm not sure I understand. The current heat equation example already uses OrdinaryDiffEq.jl and doesn't load the DifferentialEquations.jl package. From my side I haven't seen any package incompatibility issues. You can take a look at the [`Project.toml`](https://github.com/jipolanco/BSplineKit.jl/blob/master/docs/Project.toml) file used to run the examples.

**Reply by ckming102** [reply]
Sorry my bad 😅, I must have mistook the dependency in the example with my own test code that uses the full DifferentialEquations.jl.
